src/timepicker/TimePicker.ts

```diff
@@ -15,8 +29,62 @@ export const useTimePicker = createHook<TimePickerOptions, TimePickerHTMLProps>(
     compose: useBox,
     keys: TIME_PICKER_KEYS,
 
-    useProps(options, htmlProps) {
-      return htmlProps;
+    useProps(
+      options,
+      {
+        onKeyDown: htmlOnKeyDown,
+        onClick: htmlOnClick,
+        onMouseDown: htmlOnMouseDown,
+        ...htmlProps
+      },
+    ) {
+      const {
+        visible,
+        isDisabled,
+        isReadOnly,
+        show,
+        pickerId,
+        dialogId,
+        first,
+      } = options;
+
+      const onClick = () => {
+        if (isTouch()) show();
+      };
+
+      // Open the popover on alt + arrow down
+      const onKeyDown = createOnKeyDown({
+        onKey: htmlOnKeyDown,
+        preventDefault: true,
+        keyMap: event => {
+          const isAlt = event.altKey;
+
+          return {
+            ArrowDown: () => {
+              isAlt && show();
+            },
+          };
+        },
+      });
+
+      const onMouseDown = (e: React.MouseEvent) => {
+        e.stopPropagation();
+        first();
+      };
+
+      return {
+        id: pickerId,
+        role: "combobox",
+        "aria-haspopup": "dialog",
+        "aria-expanded": visible,
+        "aria-owns": visible ? dialogId : undefined,
+        "aria-disabled": ariaAttr(isDisabled),
+        "aria-readonly": ariaAttr(isReadOnly),
+        onKeyDown: callAllHandlers(htmlOnKeyDown, onKeyDown),
+        onClick: callAllHandlers(htmlOnClick, onClick),
+        onMouseDown: callAllHandlers(htmlOnMouseDown, onMouseDown),
+        ...htmlProps,
+      };
```

Comment:
This logic is very similar to DatePicker 🤔 let's refactor it to a common component.

Reply:
I am thinking the same, We need to extract this to a separate component as Picker and use it in other places.
I will do it tomorrow.

Similar Behaviours,
DatePicker - TimePicker
DatePickerTrigger - TimePickerTrigger
DatePickerContent - TimePickerContent.

They all work towards the same goal, makes combobox.

Reply:
Yeah better we extract it... let's name it "PickerBase" Picker might be confusing or refer to a concrete component

Reply:
Got it !!